### ISSUE TITLE
Proposing rolling releases

### ISSUE THREAD

**Issue by vaibhav-kaushal** [qcubed/qcubed#1150]
I think we wait too long before shipping every single feature. Version 3.0 took almost 2 years. Thanks to @spekary for the hard work there but I think we need to make the releases smaller. For example, with Redis support, we can release version 3.0.2 (or whatever the next version is). Then in 3.1 we introduce namespacing and a proper PSR-4 style loading and so on.

The codebase for everyone goes on updating slowly and steadily and we catch more bugs and move faster in general. Shall we do it?

**Comment by kukrik** [comment]
It would be very good, you can work for on the fly! 

Sure, what think @spekary?

**Comment by matthiaz** [comment]
At my day job we also work with a fixed release schedule.
Every second tuesday of the month is our release day.

The biggest problem is managing what goes into the release, and what is
moved to the next release.

So you need someone to manage that...


2017-01-12 17:38 GMT+01:00 Tiit Papp <notifications@github.com>:

**Comment by spekary** [comment]
Well, this IS an open source project with not  a lot of people working on it. JQuery UI took 2 years to release a major version, and they have hundreds of people. 

I don't see a problem with more often periodic releases. One of the goals of 3.0 was to get to the point where we could do that. 3.0 was such a major change.

As long as we move forward with requiring unit tests as much as possible, and letting a release settle for a couple of weeks, we can bump the version. That is not a problem.

We need to limit the number of new features for the dot versions. We already have a big change in 3.1 -- the new way models work. If we want to include Redis support, that would delay 3.1. But if its important to get that in sooner, then by all means add it to 3.1, and we can stop there for 3.1 and start on 3.2.

So, basically, the idea of having a faster release schedule is to simply limit the number of changes in each release.

**Comment by vaibhav-kaushal** [comment]
@spekary Exactly. 2.2 and 2.1 were released faster. Where can I see the change related to Models?

**Comment by spekary** [comment]
See the v3.1 branch.
